internal/gencrypto/token.go

```diff
@@ -0,0 +1,53 @@
+package gencrypto
+
+import (
+	"net/url"
+	"os"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/pkg/errors"
+)
+
+func LocalJWT(cluster_id string) (string, error) {
+	key, ok := os.LookupEnv("EC_PRIVATE_KEY_PEM")
+	if !ok || key == "" {
+		return "", errors.Errorf("EC_PRIVATE_KEY_PEM not found")
+	}
+	return LocalJWTForKey(cluster_id, key)
+}
+
+func LocalJWTForKey(cluster_id string, private_key_pem string) (string, error) {
+	priv, err := jwt.ParseECPrivateKeyFromPEM([]byte(private_key_pem))
+	if err != nil {
+		return "", err
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodES256, jwt.MapClaims{
+		"cluster_id": cluster_id,
+	})
+
+	tokenString, err := token.SignedString(priv)
+	if err != nil {
+		return "", err
+	}
+
+	return tokenString, nil
+}
+
+func SignURL(urlString string, cluster_id string) (string, error) {
```

Comment:
@filanov does this look like a useful "API" for whatever is going to be putting the urls into the CRs?

I'd expect the users of this to be doing something like this to get the url they pass in here:
https://github.com/openshift/assisted-service/blob/1434a2dc58134c776b5ba5eb46f6a59d8c115081/internal/isoeditor/rhcos.go#L67-L78

Reply:
If i understand correctly then this is going to be the URL to the image right?

Reply:
The input will be the url without the token auth param, the return value will be the url with the token.

The test will probably do a better job of explaining it than I can https://github.com/openshift/assisted-service/pull/1347/files#diff-ae97004f8417f6563b06969488802723b52801c74617b5afb4c09bd8a275e698R76-R87